Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -407,44 +407,39 @@ FullDiagnostics::AddRZModesToDiags (int lev)
     // E
     for (int dim=0; dim<3; dim++){
         // 3 components, r theta z
-        m_all_field_functors[lev][icomp] =
+        m_all_field_functors[lev][icomp++] =
```

Comment:
Although formally correct, this can be quite confusing for our average developer.

Most people will not immediately see that `icomp` is increased after this line, which will lead to subtle bugs at the tail of the loop over time.

Reply:
What about removing `icomp` completely and using `push_back` ?

Reply:
That works well :+1: